src/crystal/syntax_highlighter.cr

```diff
@@ -0,0 +1,220 @@
+require "html"
+require "compiler/crystal/syntax/**"
+
+abstract class Crystal::SyntaxHighlighter
+  # Parses *code* as Crystal source code and highlights it.
+  def highlight(code : String)
+    lexer = Lexer.new(code)
+    lexer.comments_enabled = true
+    lexer.count_whitespace = true
+    lexer.wants_raw = true
+
+    highlight_normal_state lexer
```

Comment:
I am confused by this call here because it returns nil. So this whole method returns nil. So this method seems both broken and unused?

Reply:
This method is the entry point for the entire parser flow. It doesn't return anything because the output is appended to an IO that is stored as an instance variable. See the implementation of the homonymous class methods for how it is used.

Reply:
Hmm I think the API docs for this method will have to be clearer then.
```crystal
# Parses *code* as Crystal source code and highlights it.
```
Highlights it and puts it where exactly?

And maybe this should direct people to mainly see "you should probably be using the class method" or something